pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -126,6 +126,28 @@ public boolean isFinal() {
       return false;
     }
   }
+
+  private class SegmentFileAndOffset implements Closeable {
+    final String _segmentFile;
+    final long _offset;
+    SegmentFileAndOffset(String segmentFile, long offset) {
+      _segmentFile = segmentFile;
+      _offset = offset;
+    }
+    public long getOffset() {
+      return _offset;
+    }
+
+    public String getSegmentFile() {
+      return _segmentFile;
+    }
+
+    @Override
+    public void close() {
```

Comment:
I wouldn't use `close` for this, since it's not really releasing a resource. `deleteSegmentFile` is a better name for this.